--- src/frontend/query-datatypes.c~	2012-07-09 17:57:04.000000000 +0200
+++ src/frontend/query-datatypes.c	2013-04-19 22:08:49.819208617 +0200
@@ -21,6 +21,8 @@
 #include <stdlib.h>
 #include <glib.h>
 #include <string.h>
+#include <unistd.h>
+
 
 #include "query-datatypes.h"
 #include "query-intl.h"
@@ -918,6 +920,7 @@
 
 fs_binding *fs_binding_join(fs_query *q, fs_binding *a, fs_binding *b, fs_join_type join)
 {
+    usleep(250); /* Inserted by Kjetil Kjernsmo for a experiment */
     if (a == NULL) {
         return fs_binding_copy(b);
     }
